src/index.js

```diff
@@ -82,6 +82,24 @@ export default class MarkersPlugin extends UICorePlugin {
     return true
   }
 
+  /*
+   * Clear all existing markers
+   */
+  clearMarkers() {
+    this._markers.forEach(marker => {
```

Comment:
I think we could introduce another method and use that here (and above):

```javascript
_removeInternalMarker(internalMarker) {
    internalMarker.$marker.remove()
    internalMarker.emitter.off("timeChanged", internalMarker.timeChangedHandler)
    if (internalMarker.$tooltipContainer) {
      internalMarker.$tooltipContainer.remove()
    }
    if (internalMarker.tooltipChangedHandler) {
      internalMarker.emitter.off("tooltipChanged", internalMarker.tooltipChangedHandler)
    }
    internalMarker.onDestroy()
    this._markers.splice(this._markers.indexOf(internalMarker), 1)
}
```
(this made me realise that line was actually missing before)